modules/auxiliary/admin/mssql/mssql_escalate_dbowner.rb

```diff
@@ -17,19 +18,25 @@ def initialize(info = {})
       },
       'Author'         => [ 'nullbind <scott.sutherland[at]netspi.com>'],
       'License'        => MSF_LICENSE,
-      'References'     => [[ 'URL','http://technet.microsoft.com/en-us/library/ms188676(v=sql.105).aspx']]
+      'References'     => [[ 'URL','http://technet.microsoft.com/en-us/library/ms188676(v=sql.105).aspx']],
+      'SessionTypes'   => %w[MSSQL]
     ))
   end
 
   def run
     # Check connection and issue initial query
-    print_status("Attempting to connect to the database server at #{rhost}:#{rport} as #{datastore['USERNAME']}...")
-    if mssql_login_datastore
-      print_good('Connected.')
+    if session
+      set_session(session.client)
+      print_good("Using session #{datastore['SESSION']}.")
     else
-      print_error('Login was unsuccessful. Check your credentials.')
-      disconnect
-      return
+      print_status("Attempting to connect to the database server at #{rhost}:#{rport} as #{datastore['USERNAME']}...")
```

Comment:
Thoughts on updating this to only be called when we don't have a session? As we won't actually be logging in.

```suggestion
      print_status("Attempting to connect to the database server at #{rhost}:#{rport} as #{datastore['USERNAME']}...") unless session
```

Reply:
There is a few instances of this that will need updated if we go ahead with this change 👍

Reply:
I believe this should already be handled by the if/else logic it's wedged in the middle of

Reply:
Whoops, with the comment above I misread the indentations, my bad 😄